packages/captp/src/atomics.js

```diff
@@ -1,6 +1,6 @@
 /// <reference types="ses"/>
 
-const { details: X } = assert;
+const { details: X, Fail } = assert;
```

Comment:
I would favor a lower-case name since this is not a constructor. I feel the same way about `X` but it’s at least a local alias. I recognize that’s a difficult proposition since we already have an `assert.fail`. Since this throws, perhaps consider `raise` (nodding to Python) or `failWithDetails` (aliased locally to `fail` or `Fail`).

I might be convinced if we’re going to establish a convention to reserve BactrianCase for both constructors and template functions, but precedent of `String.raw` seems like a fixed point that would never be consistent.

Reply:
TIL "BactrianCase". When googling it just now, my favorite comment (at https://news.ycombinator.com/item?id=965405 ) is

> I've known it as lowerCamel and UpperCamel

Reply:
OTOH, https://en.wikipedia.org/wiki/Bactrian_camel#/media/File:2011_Trampeltier_1528.JPG is memorable ;)

Reply:
> which I will hold weakly because it’s a naming preference.

Thanks for noting that. I have weak agreements with your criticism, counter-arguments for your concrete suggestions, and no better alternative to offer.

   * `failWithDetails` - way too long. For this to be pleasant to use for what we want people to use it for, the template tag name must remain small. Indeed, I don't find `failWithDetails` more pleasant than the `assert.fail(X...)` pattern it would replace.
   * `raise` - we're already using this term to mean something related but different. We're not using it a lot, so we could change this too. But then we'd need a new name for that concept. Frankly, the concept currently called `raise` is the better meaning for `raise`.

So, since your naming preference is weak, I'm resolving this without renaming `Fail`.

Reply:
For what it’s worth, people who like to be right instead of awesome just call them camelCase and PascalCase.